ic-http-outcall/proxy-canister/src/lib.rs

```diff
@@ -0,0 +1,258 @@
+use std::cell::{OnceCell, RefCell};
+use std::collections::HashMap;
+use std::sync::atomic::{AtomicU64, Ordering};
+
+use candid::Principal;
+use ic_canister::{
+    generate_idl, init, query, update, virtual_canister_call, Canister, Idl, PreUpdate,
+};
+use ic_exports::ic_cdk::api::management_canister::http_request::CanisterHttpRequestArgument;
+use ic_exports::ic_kit::ic;
+use ic_http_outcall_api::{InitArgs, RequestArgs, RequestId, ResponseResult};
+
+static IDS_COUNTER: AtomicU64 = AtomicU64::new(0);
+
+/// Canister to perform non-replicated http requests.
+#[derive(Clone, Canister)]
+#[canister_no_upgrade_methods]
+pub struct HttpProxyCanister {
+    #[id]
+    principal: Principal,
+}
+
+impl PreUpdate for HttpProxyCanister {}
+
+impl HttpProxyCanister {
+    /// Canister initialization.
+    #[init]
+    pub async fn init(&mut self, args: InitArgs) {
+        ALLOWED_PROXY.with(move |cell| {
+            cell.get_or_init(move || args.allowed_proxy);
+        });
+    }
+
+    /// Perform the non-replicated http outcall.
+    ///
+    /// The `args.callback_name` will be called, when the a reply will be ready.
+    #[update]
+    pub fn http_outcall(&mut self, args: RequestArgs) -> RequestId {
```

Comment:
As this is an unprotected update endpoint, this canister will work for free for every caller without any attempts to authenticate.